packages/cozy-doctypes/src/Folder.js

```diff
@@ -0,0 +1,54 @@
+const Application = require('./Application')
+const CozyFile = require('./File')
+
+/**
+ * Class representing the folder model.
+ * @extends CozyFile
+ */
+class CozyFolder extends CozyFile {
+  static async getReferencedFolders(client, reference) {
+    const { included } = await client
+      .collection(CozyFile.doctype)
+      .findReferencedBy({
+        _type: Application.doctype,
+        _id: reference
+      })
+    return included.filter(
+      folder => !/^\/\.cozy_trash/.test(folder.attributes.path)
+    )
+  }
+
+  static async getOrCreateFolderWithReference(client, path, reference) {
+    const existingFolders = await CozyFolder.getReferencedFolders(
+      client,
+      reference
+    )
+
+    if (existingFolders.length) return existingFolders[0]
+
+    const collection = client.collection(CozyFile.doctype)
```

Comment:
I know this is in the Drive code, but we should not use `collection()` since it bypasses all the cozyclient `links`. We should use `find` ?

Reply:
I am not sure using find allows us to use then `ensureDirectoryExists`. Could you confirm ?